extensions/typescript-language-features/src/utils/versionManager.ts

```diff
@@ -88,13 +112,38 @@ export class TypeScriptVersionManager extends Disposable {
 					await this.workspaceState.update(useWorkspaceTsdkStorageKey, true);
 					const tsConfig = vscode.workspace.getConfiguration('typescript');
 					await tsConfig.update('tsdk', version.pathLabel, false);
-					this.updateForPickedVersion(version);
+					this.updateActiveVersion(version);
 				},
 			};
 		});
 	}
 
-	private updateForPickedVersion(pickedVersion: TypeScriptVersion) {
+	public async promptUseWorkspaceTsdk(): Promise<void> {
+		const workspaceVersion = this.versionProvider.localVersion;
```

Comment:
Can you please also make sure the user is not prompted if typescript is installed locally but `typescript.tsdk` is not set

Reply:
It follows the desired behavior in local testing. The check to see whether the prompt should runs is checking for the tsdk setting in configuration [here](https://github.com/microsoft/vscode/pull/95566/files#diff-4a3e3751c6361bafa677d24e386b3cfdR162). The intent is that `promptUseWorkspaceTsdk ` should never be called in situations where the user shouldn't be prompted. Confusingly, `promptUseWorkspaceTsdk ` reads the local version from the `versionProvider` to get a reference to the instance of `TypeScriptVersion`. `configuation.localTsdk` is a string, which isn't enough to set the new TS version AFAIK